cache.js

```diff
@@ -0,0 +1,136 @@
+var exports = module.exports = {};
+var https = require("https");
+var fs = require('fs');
+
+function downloadFile(url, fileName) {
+
+	var file = fs.createWriteStream(fileName);
+	https.get(url, function(res) {
+		res.pipe(file);
+		file.on('finish', function() {
+			file.close();
+		});
+
+	}).on('error', function() {
+			console.log("Could not write file");
+
+	});
+}
+
+function cacheFiles(fileList, oldList, token) {
+
+	for (var i = 0; i < fileList.length; i++) {

```

Comment:
This won't scale - needs to be re-worked to use callbacks, i.e. download a file, callback when download complete, download next file and so on. Otherwise the network and device will be swamped with multiple concurrent requests. Or use Promises.